src/esp/physics/PhysicsObjectBase.h

```diff
@@ -427,13 +427,17 @@ class PhysicsObjectBase : public Magnum::SceneGraph::AbstractFeature3D {
   core::Configuration::ptr getUserAttributes() const { return userAttributes_; }
 
   /**
-   * @brief This function will completely overwrite this object's
-   * user-defined attributes.
+   * @brief This function will overwrite or merge this object's existing
+   * user-defined attributes with @p attr , based on value of @p merge
    * @param attr A ptr to the user defined attributes specified for this object.
    * merge into them.
    */
-  void setUserAttributes(core::Configuration::ptr attr) {
-    userAttributes_ = std::move(attr);
+  void setOrMergeUserAttributes(core::Configuration::ptr attr, bool merge) {
```

Comment:
[Not a fan of booleans in APIs](https://ariya.io/2011/08/hall-of-api-shame-boolean-trap), why can't it be two different functions? Shorter to type, easier to understand, and the two functions don't really have any common code to begin with so putting them together doesn't make anything simpler.

Reply:
I had this because I wanted to support a process where I was not sure which call would be first.  That is probably not a reasonable concern, really - there's a call right after the object is created, and then another when it is placed.  There's not going to be confusion about this.